### ISSUE TITLE
Removing warnings from the Sphinx build

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#491]
At present the Travis Sphinx build ignores warnings generated by Sphinx. This is not ideal, since it would be good if it could fail on warnings. However, the current build generates plenty of warnings which are not easy to remove. These fall into two camps

1. formatting issues from the docstrings in `sherpa.utils.testing.SherpaTestCase`, which are inherited from 

2. complaints about duplicate target names (for footnotes) from docstrings in `sherpa.ui.utils.Session` and `sherpa.astro.ui.utils.Session` which I *believe* are because of the "funky" way we use these symbols (there are class methods which are re-exported as functions from `sherpa[.astro].ui`)

One way to address the first point would be to update the tests using the `SherpaTestCase` class to us pytest instead, so we could remove the class itself. or we could just manually over-ride the problem routines and write out own docstrings.

I have no idea about the second point.

**Comment by hamogu** [comment]
I believe (but I might be wrong) that the second point is real, not just "funky". It's because the references are always called 1,2,3 as in `[1]`. I believe they have to be manually fixed to unique labels like`[Cash1979]` (that's hoping that no paper is cited twice). Sphinx is not made to have a separate reference section in every function with labels 1,2,3; sphinx citations are meant to be global: http://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#citations

**Comment by DougBurke** [comment]
The first point is (finally) addressed in #1205

**Comment by hamogu** [comment]
After some further investigation I can confirm what I said in https://github.com/sherpa/sherpa/issues/491#issuecomment-618732309: Every reference in sphinx is global and thus we cannot use labels like [1], [2], because we simply have too many of them. The intention of Sphinx is more to act like LaTeX, where `\citep{Bouke2013}` will give you the same reference throughout. You *could* use `\citep{1}`, but nobody does because the "1" would the same reference throughout the text. 

That's not what we do - we use [1] as a throwaway reference in out API and there are dozens of functions wit ha reference called `[1]`, at least a dozen of them refer to ds9 and another 10 or so to the OGIP memo. If we want to keep per-function references, we have to be more create with reference names to makes them unique, e.g. with names as "ds9", "DS9", "ref1", "foot1", etc. 
That's also not ideal because it's inconsistent. 

However, even if one does that (I made branch where I tried), that **still** is not enough, because the same docs are  often build in *tow different locations*:
- https://sherpa.readthedocs.io/en/4.14.1/ui/api/sherpa.astro.ui.calc_kcorr.html
- https://sherpa.readthedocs.io/en/4.14.1/ui/api/sherpa.astro.ui.utils.Session.html#sherpa.astro.ui.utils.Session.calc_kcorr

Are two different web-pages, which are build from the same docstring - once as a method on the Session object and once as a function of `sherpa.astro.ui`. Because they are build from the same docstring, they necessarily have the same reference name 0 ans thus still trigger the following warning:

`/Users/guenther/code/sherpa/sherpa/astro/ui/utils.py:docstring of sherpa.astro.ui.utils.Session.calc_kcorr:69: WARNING: duplicate citation k2, other instance in /Users/guenther/code/sherpa/docs/ui/api/sherpa.astro.ui.calc_kcorr.rst`

So, within sphinx, there is no way of solving this, because we are not using sphinx that way it's meant to be used (in at least two aspects). It works, but we get the warnings.

Yet, these warnings are annoying me enough (in #1382 I missed relevant sphinx warning several times because they were burried under all this useless noise), that I'll suggest some ways to at least reduce the number of warnings - see forthcoming PRs.
